elastic/kibana
# [Dashboard] [Controls] Fix unexpected suggestions for text/keyword multi-fields

Closes https://github.com/elastic/kibana/issues/143919

## Summary

This PR makes it so that the search string is no longer unnecessarily escaped for `text / keyword` [multi-fields](https://www.elastic.co/guide/en/elasticsearch/reference/current/text.html#text-multi-fields) when building the options list suggestions.

For `keyword` fields, escaping the search string **is** necessary since the `terms` aggregation treats the `include` property as a regular expression. However, in order to support case-insensitive search for `text / keyword` multi-fields, we use the `match_phrase_prefix` filter instead - this query will tokenize the search string as part of the text analysis (as described in the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis.html)), which does not require escaping. Hence, why you would get unexpected results when a `text / keyword` multi-field data contained escapable characters.

![image](https://user-images.githubusercontent.com/8698078/201755604-d2396cdb-50ec-415a-bbed-655c85bde3ad.png)

**How to Test**
1. Create an index with that has a field mapped as a `text / keyword` multi-field with some documents where this field has punctuation that would normally be escaped - for example, download 
[testPunctuation.csv](https://github.com/elastic/kibana/files/10006642/testPunctuation.csv) and upload it to Kibana, ensuring you update the auto-mapping to be:
    ```json
    {
      "id": {
        "type": "long"
      },
      "sentence": {
        "type": "text",
        "fields": {
          "keyword": {
            "type": "keyword"
          }
        }
      }
    }
    ```
2. Create a control on this `text / keyword` multi-field  - for example, on the `sentence.keyword` field if using the provided sample data
3. Ensure that various searches return reasonable suggestions. Please note that using a `match_phrase_prefix` for an auto-complete functionality, as we are doing in this instance, **can** still give some slightly unexpected results - this is described in more detail in the relevant [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query-phrase-prefix.html#match-phrase-prefix-autocomplete). However, since case-insensitive search is desired, this is a necessary trade-off for supporting this.

### Checklist



### For maintainers
